### PR TITLE
K8SPG-359 Added support for custom user secret name

### PR DESCRIPTION
**Pull request by vishalanarase**
[K8SPG-359](https://jira.percona.com/browse/K8SPG-359) [Powered by Pull Request Badge](https://pullrequestbadge.com/?utm_medium=github&utm_source=percona&utm_campaign=badge_info)

**CHANGE DESCRIPTION**
---
**Problem:**
*There wasn't support for a custom user secret name.  There was fix pattern to generate the name*

**Cause:**
*When the user gives cr name of more than 63 characters secret isn't created so can't access the creds*

**Solution:**
*This PR adds support for custom user secret name*

**CHECKLIST**
---
**Jira**
- [x] Is the Jira ticket created and referenced properly?
- [ ] Does the Jira ticket have the proper statuses for documentation (`Needs Doc`) and QA (`Needs QA`)?
- [ ] Does the Jira ticket link to the proper milestone (Fix Version field)?

**Tests**
- [x] Is an E2E test/test case added for the new feature/change?
- [x] Are unit tests added where appropriate?

**Config/Logging/Testability**
- [x] Are all needed new/changed options added to default YAML files?
- [x] Are the manifests (crd/bundle) regenerated if needed?
- [ ] Did we add proper logging messages for operator actions?
- [ ] Did we ensure compatibility with the previous version or cluster upgrade process?
- [ ] Does the change support oldest and newest supported PG version?
- [ ] Does the change support oldest and newest supported Kubernetes version?

### CODE REVIEW

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-442/a91303e53/init-deploy.log) |
| demand-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-442/a91303e53/demand-backup.log) |
| scheduled-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-442/a91303e53/scheduled-backup.log) |
| start-from-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-442/a91303e53/start-from-backup.log) |
| monitoring | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-442/a91303e53/monitoring.log) |
| telemetry-transfer | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-442/a91303e53/telemetry-transfer.log) |
| We run 6 out of 6|

commit: https://github.com/percona/percona-postgresql-operator/pull/442/commits/a91303e536bdc382abb2def5a8868bee46480e8a
image: `perconalab/percona-postgresql-operator:PR-442-a91303e53`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-442/e1615001a/init-deploy.log) |
| demand-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-442/e1615001a/demand-backup.log) |
| scheduled-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-442/e1615001a/scheduled-backup.log) |
| start-from-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-442/e1615001a/start-from-backup.log) |
| monitoring | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-442/e1615001a/monitoring.log) |
| telemetry-transfer | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-442/e1615001a/telemetry-transfer.log) |
| We run 6 out of 6|

commit: https://github.com/percona/percona-postgresql-operator/pull/442/commits/e1615001a2ccd6cb8d13adf79857f9e12da8674e
image: `perconalab/percona-postgresql-operator:PR-442-e1615001a`

File: deploy/bundle.yaml [hunk 1]
```diff
@@ -30312,7 +30324,7 @@ spec:
           value: "true"
         - name: DISABLE_TELEMETRY
           value: "false"
-        image: perconalab/percona-postgresql-operator:main
+        image: perconalab/percona-postgresql-operator:k8spg-359-improvement-user-defined-secret-name
```
Comment (hors) [inline deploy/bundle.yaml:30327]:
```suggestion
        image: perconalab/percona-postgresql-operator:main
```

File: deploy/operator.yaml [hunk 2]
```diff
@@ -42,7 +42,7 @@ spec:
           value: "true"
         - name: DISABLE_TELEMETRY
           value: "false"
-        image: perconalab/percona-postgresql-operator:main
+        image: perconalab/percona-postgresql-operator:k8spg-359-improvement-user-defined-secret-name
```
Comment (hors) [inline deploy/operator.yaml:45]:
```suggestion
        image: perconalab/percona-postgresql-operator:main
```

File: config/manager/kustomization.yaml [hunk 3]
```diff
@@ -11,4 +11,4 @@ commonLabels:
 images:
 - name: postgres-operator
   newName: perconalab/percona-postgresql-operator
-  newTag: main
+  newTag: k8spg-359-improvement-user-defined-secret-name
```
Comment (hors) [inline config/manager/kustomization.yaml:14]:
```suggestion
  newTag: main
```

File: config/bundle/kustomization.yaml [hunk 4]
```diff
@@ -7,4 +7,4 @@ resources:
 images:
 - name: postgres-operator
   newName: perconalab/percona-postgresql-operator
-  newTag: main
+  newTag: k8spg-359-improvement-user-defined-secret-name
```
Comment (hors) [inline config/bundle/kustomization.yaml:10]:
```suggestion
  newTag: main
```

**Review by hors — CHANGES_REQUESTED** [review]

File: testing/kuttl/e2e/password-change/12--custom-user-secret-name.yaml
Comment (hors) [inline testing/kuttl/e2e/password-change/12--custom-user-secret-name.yaml:1]: Please move test to /e2e-tests/tests/

**Review by hors — CHANGES_REQUESTED** [review]

**Comment by egegunes** [comment]
@vishalanarase you need to put this field to our custom resource: https://github.com/percona/percona-postgresql-operator/blob/main/pkg/apis/pg.percona.com/v2beta1/perconapgcluster_types.go#L38

and then you need to map this field to crunchy CR in: https://github.com/percona/percona-postgresql-operator/blob/main/percona/controller/pgcluster/controller.go#L144-L213

otherwise you can't use this field through our custom resources

File: config/bundle/kustomization.yaml (hunk 4 above)
Comment (vishalanarase) [reply config/bundle/kustomization.yaml]: Done

File: config/manager/kustomization.yaml (hunk 3 above)
Comment (vishalanarase) [reply config/manager/kustomization.yaml]: Done

File: deploy/bundle.yaml (hunk 1 above)
Comment (vishalanarase) [reply deploy/bundle.yaml]: Done

File: deploy/operator.yaml (hunk 2 above)
Comment (vishalanarase) [reply deploy/operator.yaml]: Done

**Comment by vishalanarase** [comment]
> @vishalanarase you need to put this field to our custom resource: https://github.com/percona/percona-postgresql-operator/blob/main/pkg/apis/pg.percona.com/v2beta1/perconapgcluster_types.go#L38
> 
> and then you need to map this field to crunchy CR in: https://github.com/percona/percona-postgresql-operator/blob/main/percona/controller/pgcluster/controller.go#L144-L213
> 
> otherwise you can't use this field through our custom resources

I can't see the `PostgresUserSpec` struct in your types. I see it only in the https://github.com/percona/percona-postgresql-operator/blob/main/pkg/apis/postgres-operator.crunchydata.com/v1beta1/postgres_types.go#L41

File: testing/kuttl/e2e/password-change/12--custom-user-secret-name.yaml
Comment (vishalanarase) [reply testing/kuttl/e2e/password-change/12--custom-user-secret-name.yaml:1]: Done

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-442/8270302b9/init-deploy.log) |
| demand-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-442/8270302b9/demand-backup.log) |
| scheduled-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-442/8270302b9/scheduled-backup.log) |
| start-from-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-442/8270302b9/start-from-backup.log) |
| monitoring | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-442/8270302b9/monitoring.log) |
| telemetry-transfer | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-442/8270302b9/telemetry-transfer.log) |
| We run 6 out of 7|

commit: https://github.com/percona/percona-postgresql-operator/pull/442/commits/8270302b99267052c0c2582153a88c7654d1eb6d
image: `perconalab/percona-postgresql-operator:PR-442-8270302b9`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-442/e4865f4f9/init-deploy.log) |
| demand-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-442/e4865f4f9/demand-backup.log) |
| scheduled-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-442/e4865f4f9/scheduled-backup.log) |
| start-from-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-442/e4865f4f9/start-from-backup.log) |
| monitoring | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-442/e4865f4f9/monitoring.log) |
| telemetry-transfer | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-442/e4865f4f9/telemetry-transfer.log) |
| We run 6 out of 7|

commit: https://github.com/percona/percona-postgresql-operator/pull/442/commits/e4865f4f93fbea5c729862ed0ca3630911e84076
image: `perconalab/percona-postgresql-operator:PR-442-e4865f4f9`

**Review by hors — APPROVED** [review]

**Comment by vishalanarase** [comment]
Need one more approval @egegunes can please review and approve?

**Review by egegunes — APPROVED** [review]

**Comment by hors** [comment]
@vishalanarase thank you for conrebution

**Pull request merged**